source/agent/analytics/videoGstPipeline/VideoGstAnalyzer.cc

```diff
@@ -272,12 +288,23 @@ void VideoGstAnalyzer::new_sample_from_sink (GstElement * source, gpointer data)
     owt_base::Frame outFrame;
```

Comment:
for robustness, make sure you check validity of sample and buffer before using them.